code/modules/mob/living/carbon/xenomorph/xenoprocs.dm

```diff
@@ -76,10 +78,12 @@
 	xenoinfo += xeno_status_output(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/larva], can_overwatch, TRUE, user)
 
 	var/hivemind_text = length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/hivemind]) > 0 ? "Active" : "Inactive"
+	var/xeno_tier_three_cap = CEILING(max(length(hive.xenos_by_tier[XENO_TIER_THREE]),(length(hive.xenos_by_tier[XENO_TIER_ZERO])+length(hive.xenos_by_tier[XENO_TIER_ONE])+length(hive.xenos_by_tier[XENO_TIER_TWO]))/3),1)
+	var/xeno_tier_two_cap = max(length(hive.xenos_by_tier[XENO_TIER_TWO]),length(hive.xenos_by_tier[XENO_TIER_ZERO])+length(hive.xenos_by_tier[XENO_TIER_ONE])-length(hive.xenos_by_tier[XENO_TIER_THREE]))
```

Comment:
use the TO_XENO_TIER_2_FORMULA and TO_XENO_TIER_3_FORMULA defines

Reply:
These do not produce the actual number of T2 and T3 slots; it only determines whether or not the hive can support more T2s or T3s than currently exist

Reply:
Ok todo fix shitty evolve macros